src/session/session_api.c

```diff
@@ -1947,16 +2070,19 @@ __open_session(WT_CONNECTION_IMPL *conn,
 
 	TAILQ_INIT(&session_ret->cursors);
 	TAILQ_INIT(&session_ret->dhandles);
-	/*
-	 * If we don't have one, allocate the dhandle hash array.
-	 * Allocate the table hash array as well.
-	 */
+
+	/* If we don't have one, allocate the dhandle hash array. */
 	if (session_ret->dhhash == NULL)
 		WT_ERR(__wt_calloc(session, WT_HASH_ARRAY_SIZE,
 		    sizeof(struct __dhandles_hash), &session_ret->dhhash));
 	for (i = 0; i < WT_HASH_ARRAY_SIZE; i++)
 		TAILQ_INIT(&session_ret->dhhash[i]);
 
+	/* Initialize the cursor cache hash buckets and sweep trigger. */
+	for (i = 0; i < WT_HASH_ARRAY_SIZE; i++)
+		TAILQ_INIT(&session_ret->cursor_cache[i]);
+	session_ret->cursor_sweep_countdown = WT_SESSION_CURSOR_SWEEP_COUNTDOWN;
+
```

Comment:
Unless I'm not seeing it, I think we're leaking the cursor cache array. The `dhhash` array is freed in `conn_open.c:175` and maybe this should be too?

Reply:
@sueloverso, `cursor_cache` is statically sized in the `__wt_session_impl` struct (session session.h: line 82).

Reply:
I agree, I've modified dhhash to be statically allocated in the structure.  That matches similar uses in `connection.h`.